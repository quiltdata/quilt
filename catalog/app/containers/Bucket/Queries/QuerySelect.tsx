import * as React from 'react'
import * as M from '@material-ui/core'

// import { docs } from 'constants/urls'
// import StyledLink from 'utils/StyledLink'

import * as requests from './requests'

interface QuerySelectProps {
  className: string
  queries: requests.Query[]
  onChange: (value: requests.Query | null) => void
  value: requests.Query | null
}

const useStyles = M.makeStyles((t) => ({
  header: {
    margin: t.spacing(0, 0, 1),
  },
  selectWrapper: {
    width: '100%',
  },
  select: {
    padding: t.spacing(1),
  },
}))

export default function QuerySelect({
  className,
  queries,
  onChange,
  value,
}: QuerySelectProps) {
  const classes = useStyles()

  const handleChange = React.useCallback(
    (event) => {
      onChange(queries.find((query) => query.key === event.target.value) || null)
    },
    [queries, onChange],
  )

  return (
    <div className={className}>
      <M.Typography className={classes.header} variant="body1">
        Select query
      </M.Typography>
      <M.Paper>
        <M.FormControl className={classes.selectWrapper}>
          <M.Select
            classes={{ root: classes.select }}
            disabled={!queries.length}
            labelId="query-select"
            onChange={handleChange}
            value={value ? value.key : 'none'}
          >
            <M.MenuItem disabled value="none">
              Custom
            </M.MenuItem>
            {queries.map((query) => (
              <M.MenuItem key={query.key} value={query.key}>
                {query.name}
              </M.MenuItem>
            ))}
          </M.Select>
        </M.FormControl>
      </M.Paper>
      <M.FormHelperText>
        {!queries.length && 'There are no saved queries. '}
        {/* <StyledLink href={`${docs}/advanced-usage/queries`} target="_blank">
          Refer to documentation
        </StyledLink>{' '}
        to edit or save new queries. */}
      </M.FormHelperText>
    </div>
  )
}
