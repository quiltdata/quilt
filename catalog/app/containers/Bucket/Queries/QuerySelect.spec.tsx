import * as React from 'react'
import renderer from 'react-test-renderer'

import QuerySelect from './QuerySelect'

describe('containers/Bucket/Queries/QuerySelect', () => {
  it('should render', () => {
    const tree = renderer
      .create(<QuerySelect className="" queries={[]} onChange={() => {}} value={null} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('should render with selected value', () => {
    const queries = [
      { key: 'key1', name: 'name1', url: 'url1' },
      { key: 'key2', name: 'name2', url: 'url2' },
    ]
    const tree = renderer
      .create(
        <QuerySelect
          className="class"
          queries={queries}
          onChange={() => {}}
          value={queries[1]}
        />,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
