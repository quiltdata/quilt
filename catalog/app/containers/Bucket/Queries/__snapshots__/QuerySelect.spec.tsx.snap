// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/Bucket/Queries/QuerySelect should render 1`] = `
<div
  className=""
>
  <p
    className="MuiTypography-root makeStyles-header-1 MuiTypography-body1"
  >
    Select query
  </p>
  <div
    className="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      className="MuiFormControl-root makeStyles-selectWrapper-2"
    >
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline Mui-disabled Mui-disabled MuiInputBase-formControl MuiInput-formControl"
        onClick={[Function]}
      >
        <div
          aria-disabled="true"
          aria-haspopup="listbox"
          aria-labelledby="query-select"
          className="MuiSelect-root makeStyles-select-3 MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input Mui-disabled Mui-disabled Mui-disabled"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseDown={null}
          role="button"
          tabIndex={null}
        >
          Custom
        </div>
        <input
          aria-hidden={true}
          className="MuiSelect-nativeInput"
          onAnimationStart={[Function]}
          onChange={[Function]}
          required={false}
          tabIndex={-1}
          value="none"
        />
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSelect-icon Mui-disabled"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
    </div>
  </div>
  <p
    className="MuiFormHelperText-root"
  >
    There are no saved queries. 
  </p>
</div>
`;

exports[`containers/Bucket/Queries/QuerySelect should render with selected value 1`] = `
<div
  className="class"
>
  <p
    className="MuiTypography-root makeStyles-header-1 MuiTypography-body1"
  >
    Select query
  </p>
  <div
    className="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      className="MuiFormControl-root makeStyles-selectWrapper-2"
    >
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        onClick={[Function]}
      >
        <div
          aria-haspopup="listbox"
          aria-labelledby="query-select"
          className="MuiSelect-root makeStyles-select-3 MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          role="button"
          tabIndex={0}
        >
          name2
        </div>
        <input
          aria-hidden={true}
          className="MuiSelect-nativeInput"
          onAnimationStart={[Function]}
          onChange={[Function]}
          required={false}
          tabIndex={-1}
          value="key2"
        />
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSelect-icon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
    </div>
  </div>
  <p
    className="MuiFormHelperText-root"
  />
</div>
`;
